data_diff/databases/snowflake.py

```diff
@@ -35,10 +37,23 @@ def __init__(self, *, schema: str, **kw):
         logging.getLogger("snowflake.connector.network").disabled = True
 
         assert '"' not in schema, "Schema name should not contain quotes!"
-        self._conn = snowflake.connector.connect(
-            schema=f'"{schema}"',
-            **kw,
-        )
+        if (
+            "key" in kw
+        ):  # if private keys are used for Snowflake connection, read in key from path specified and pass as "private_key" to connector.
+            with open(kw.get("key"), "rb") as key:
+                p_key = serialization.load_pem_private_key(
+                    key.read(),
+                    password=None if not kw.get("password") else kw.pop("password"),
```

Comment:
You can just do `kw.pop("password", None)`

Reply:
Thanks, done!